packages/mui-material/src/SpeedDialAction/SpeedDialAction.js

```diff
@@ -173,44 +176,81 @@ const SpeedDialAction = React.forwardRef(function SpeedDialAction(inProps, ref)
 
   const transitionStyle = { transitionDelay: `${delay}ms` };
 
-  const fab = (
-    <SpeedDialActionFab
-      size="small"
-      className={clsx(classes.fab, className)}
-      tabIndex={-1}
-      role="menuitem"
-      ownerState={ownerState}
-      {...FabProps}
-      style={{
+  const externalForwardedProps = {
+    slots,
+    slotProps: {
+      fab: FabProps,
+      ...slotProps,
+    },
+  };
+
+  const [FabSlot, fabSlotProps] = useSlot('fab', {
+    elementType: SpeedDialActionFab,
+    externalForwardedProps,
+    ownerState,
+    shouldForwardComponentProp: true,
+    className: clsx(classes.fab, className),
+    additionalProps: {
+      style: {
```

Comment:
```suggestion
      style: transitionStyle
```
No need to spread the `FabProps.style`, the `slotProps.fab.style` (from `FabProps`) will be merged by `useSlot`.